### PR TITLE
MGMT-6769: Allows the users to select the network type (SDN/OVN)

### PR DESCRIPTION
**Pull request by jkilzi**
[MGMT-6769](https://issues.redhat.com/browse/MGMT-6769)

https://user-images.githubusercontent.com/77008341/127576039-3ffc45b5-a6ae-4a39-bd13-ba76c87d7186.mp4

### CODE REVIEW

File: src/common/components/clusterConfiguration/AdvancedNetworkFields.tsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+import { InputField } from '../../components/ui';
 import { NetworkConfigurationValues } from '../../types/clusters';
-import { PREFIX_MAX_RESTRICTION } from '../../config';
-import { InputField } from '../ui';
+import { Address6 } from 'ip-address';
+import { PREFIX_MAX_RESTRICTION } from '../../config/constants';
+import { NetworkTypeControlGroup } from '../clusterWizard/networkingSteps/NetworkTypeControlGroup';
 
-const AdvancedNetworkFields: React.FC = () => {
+const AdvancedNetworkFields = () => {
```
Comment (rawagner) [inline src/common/components/clusterConfiguration/AdvancedNetworkFields.tsx:10]: not sure why the type was removed. It should actually be `React.FC<{}>`. Not a big deal though

File: src/common/components/clusterWizard/networkingSteps/NetworkTypeControlGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { Split, SplitItem, Tooltip } from '@patternfly/react-core';
+import { RadioField } from '../../ui/formik';
+import { PopoverIcon } from '../../../components/ui';
+
+export interface NetworkTypeControlGroupProps {
+  isIPv6: boolean;
+}
+
+export const NetworkTypeControlGroup = ({ isIPv6 = false }: NetworkTypeControlGroupProps) => {
```
Comment (rawagner) [inline src/common/components/clusterWizard/networkingSteps/NetworkTypeControlGroup.tsx:10]: I prefer to be consistent with typing of our components
```suggestion
export const NetworkTypeControlGroup: React.FC<NetworkTypeControlGroupProps> = ({ isIPv6 = false }) => {
```

File: src/common/components/ui/formik/RadioField.tsx [hunk 3]
```diff
@@ -3,15 +3,12 @@ import { useField } from 'formik';
 import { Radio, RadioProps } from '@patternfly/react-core';
 import { getFieldId } from './utils';
 
-const RadioField = (props: RadioProps) => {
-  const [field] = useField(props.name);
-  /* TODO(mlibra): Hotfix - verify!
-  const [field] = useField<NetworkConfigurationValues['networkingType']>({
+const RadioField = (props: Optional<RadioProps, 'id'>) => {
```
Comment (rawagner) [inline src/common/components/ui/formik/RadioField.tsx:6]:
```suggestion
const RadioField: React.FC<Optional<RadioProps, 'id'>> = (props) => {
```

File: src/common/components/clusterConfiguration/AdvancedNetworkFields.tsx (hunk 1 above)
Comment (jkilzi) [reply src/common/components/clusterConfiguration/AdvancedNetworkFields.tsx:10]:
```suggestion
const AdvancedNetworkFields: React.FC<{}> = () => {
```

**Review by jgyselov — APPROVED** [review]

**Review by rawagner — APPROVED** [review]

**Pull request merged**